lib/northbound_db.c

```diff
@@ -87,9 +87,13 @@ int nb_db_transaction_save(const struct nb_transaction *transaction,
 		goto exit;
 
 	client_name = nb_client_name(transaction->context->client);
-	/* Always record configurations in the XML format. */
+	/*
+	 * Always record configurations in the XML format, save the default
+	 * values too, as when we rollback we want them to replace non-default
+	 * values
```

Comment:
> as when we rollback we want them to replace non-default values

Not really necessary as `lyd_diff()` is capable of handling implicit defaults appropriately. But I think it's a good idea to record implicit defaults nonetheless to handle defaults that change across FRR versions. It would be good to update this doc comment to reflect that intent.

Reply:
updated